docker/jenkins/Jenkinsfile

```diff
@@ -1,46 +1,858 @@
+          }
+        }
+        stage('BNN Sanity - KV260_SOM') {
+          when {
+            // beforeAgent set to 'true' to prevent an offline agent hanging the stage
+            beforeAgent true
+            expression { return (env.KV260_ONLINE == 'true' && params['sanity'] && env.BNN_BUILD_SANITY == 'SUCCESS') }
+          }
+          agent {
+            label 'finn-kv260'
+          }
+          environment {
+            BOARD = 'KV260_SOM'
+            USER_CREDENTIALS = credentials('user-ubuntu-credentials')
+          }
+          steps {
+            catchError(stageResult: 'FAILURE') {
+              script {
+                // Clean any files from a previous run
+                sh "echo $USER_CREDENTIALS_PSW | sudo -S rm -rf ${env.BOARD}*"
+
+                // Get the test files
+                unstash name: "sanity_${env.BOARD}_zip"
+                sh "unzip -o ${env.BOARD}.zip"
+
+                dir(env.BOARD) {
+                  // Get the scripts necessary for running hw tests
+                  unstash name: 'bnn_test_files'
+
+                  // Create test script
+                  createTestScript(env.BOARD, env.BOARD, "sanity_bnn_test_hw_${env.BOARD}")
+
+                  // Use an env variable to help collect test results later in pipeline
+                  env.SANITY_BNN_TEST_KV260_SOM = "SUCCESS"
+
+                  // Execute the script as the root user - needed for zynq platforms
+                  sh 'echo $USER_CREDENTIALS_PSW | sudo -S ./run-tests.sh'
                 }
+              }
+            }
+          }
+           post {
+            always {
+              // Get test result file and delete test files on the board
+              dir(env.BOARD) {
+                // Collect the results file on the slave node by stashing
+                stash name: "xml_sanity_bnn_test_${env.BOARD}", includes: "sanity_bnn_test_hw_${env.BOARD}.xml,sanity_bnn_test_hw_${env.BOARD}.html"
+              }
             }
-        }, fifthBranch: {
-            stage('Fpgadataflow transformations and simulations') {
-                dir("${env.WORKSPACE}") {
-                sh("bash run-docker.sh python setup.py test --addopts -mfpgadataflow")
+          }
+        }
+      }
+    }
+    stage('End2end - Run Hardware Tests') {
+      parallel {
+        stage('BNN end2end - U250') {
+          when {
+            // beforeAgent set to 'true' to prevent an offline agent hanging the stage
+            beforeAgent true
+            expression { return (env.ALVEO_HOST_ONLINE == 'true' && params['end2end'] && env.BNN_BUILD_U250 == 'SUCCESS') }
+          }
+          agent {
+            label 'finn-u250'
+          }
+          environment {
+            BOARD = 'U250'
+          }
+          steps {
+            catchError(stageResult: 'FAILURE') {
+              script {
+                // Clean any files from a previous run
+                sh "rm -rf ${env.BOARD}*"
```

Comment:
Could use the cleanPreviousBuildFiles() method

Reply:
Fixed